### PR TITLE
Documented gymbal lock cases

### PR DESCRIPTION
**Pull request by evbernardes**
#### References to other Issues or PRs
Follow up to #24604 and #24605.

#### Brief description of what is fixed or changed
Modified docstring of `Quaternion.to_euler` to explain how degenerate cases are dealt with.

As another minor edit, replaced `sympify(0)` with `S.Zero`.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/algebras/quaternion.py [hunk 1]
```diff
@@ -442,6 +442,9 @@ def to_euler(self, seq, angle_addition=True, avoid_square_root=False):
         in the sequence given by `seq`. This implements the method described
```
Comment (sylee957) [inline sympy/algebras/quaternion.py:442]:
```suggestion
        in the sequence given by ``seq``. This implements the method described
```

**Comment by sylee957** [comment]
I recognize that the some math is broken in 
https://output.circle-artifacts.com/output/job/e4dc125b-11b0-4b99-8af4-1d5cf1cd664a/artifacts/0/doc/_build/html/modules/algebras.html#sympy.algebras.Quaternion.to_euler
can you fix the problems with set notation and backticks and the formula in `avoid_square_root`?

**Comment by evbernardes** [comment]
> I recognize that the some math is broken in https://output.circle-artifacts.com/output/job/e4dc125b-11b0-4b99-8af4-1d5cf1cd664a/artifacts/0/doc/_build/html/modules/algebras.html#sympy.algebras.Quaternion.to_euler can you fix the problems with set notation and backticks and the formula in `avoid_square_root`?

Oops indeed, sorry about that. I think it's fixed now, added double backticks everywhere and fixed the call to the math environment.

**Comment by evbernardes** [comment]
Ok, I think this is good now. Did a final commit adding double backticks to `from_euler` too.

**Pull request merged**